graphql/appointment/mutations.ts

```diff
@@ -202,7 +203,44 @@ export class MutateAppointmentResolver {
         const appointment = await getLecture(appointmentId);
         await hasAccess(context, 'Lecture', appointment);
         await prisma.lecture.update({ data: { isCanceled: true }, where: { id: appointment.id } });
-        // * Send notification here
+
+        const student = await getStudent(context.user.studentId);
+        const language = 'de-DE';
+
+        if (appointment.appointmentType === lecture_appointmenttype_enum.group) {
+            const subcourse = await prisma.subcourse.findFirst({ where: { id: appointment.subcourseId }, include: { course: true } });
+            const participants = await prisma.subcourse_participants_pupil.findMany({ where: { subcourseId: subcourse.id } });
+
+            for await (const participant of participants) {
+                const pupil = await getPupil(participant.pupilId);
```

Comment:
This can be merged into the participants query by adding a `include: { pupil: true }`